src/encoded/types/shared_calculated_properties.py

```diff
@@ -284,7 +284,17 @@ class CalculatedVisualize:
     def visualize(self, request, hub, accession, assembly, status, files):
         hub_url = urljoin(request.resource_url(request.root), hub)
         viz = {}
-        for assembly_name in assembly:
+        vis_assembly = set()
+        viewable_file_formats = ['bigWig', 'bigBed']
+        viewable_file_status = ['released','in progress']
+
+        for properties in files:
```

Comment:
This can be updated to a set comprehension expression.  When to do this?  When creating a list, set, or dict for one time use using ADD conditions

Here we're filtering the files based on some simple AND conditions, putting them a set, and then consuming that set when creating the viz dict.   
```
vis_assembly = {
    properties['assembly']
    for properties in files
    if properties.get('file_format') in viewable_file_formats
    if properties.get('status') in viewable_file_status
    if 'assembly' in properties
}
```
Python's built in syntax for this common programming pattern can run faster with less memory.

I also add dict.get(key).  This is safer than dict[key] because if the key doesn't exist the get method return None.  And there is nothing wrong with checking 'if None in list'.  So long as None is not in the list!